bin/metrics-pods.rb

```diff
@@ -0,0 +1,59 @@
+#! /usr/bin/env ruby
+#
+#   pod-metrics
+#
+# DESCRIPTION:
+#   Will give pod counts from all the exposed services
+#
+# OUTPUT:
+#   metric data
+#
+# PLATFORMS:
+#   Linux
+#
+# DEPENDENCIES:
+#   gem: sensu-plugin
+#   gem: kube-client
+#
+# USAGE:
+#
+# NOTES:
+#
+# LICENSE:
+#   Copyright 2016 Chris McFee <cmcfee@kent.edu>
+#   Released under the same terms as Sensu (the MIT license); see LICENSE
+#   for details.
+require 'sensu-plugins-kubernetes/cli'
+require 'sensu-plugin/metric/cli'
+require 'uri'
+
+class PodsMetrics < Sensu::Plugin::Metric::CLI::Graphite
+  @options = Sensu::Plugins::Kubernetes::CLI.options.dup
+
+  def run
+    config[:scheme] = "#{URI(config[:api_server]).host}.pods"
+    count = {}
+    client = Sensu::Plugins::Kubernetes::CLI.new.client
+    services = client.get_services
+    services.each do |s|
+      selector_key = []
+      count[s.metadata.name] = 0
+      services.delete(s.metadata.name)
+      s.spec.selector.to_h.each do |k, v|
+        selector_key << "#{k}=#{v}"
+      end
+      pod = nil
+      begin
+        pod = client.get_pods(label_selector: selector_key.join(',').to_s)
+      rescue
+        puts 'There was an error'
```

Comment:
this should be more descriptive and raise the actual error along with using warn or critical